### ISSUE TITLE
"RangeError: Maximum call stack size exceeded" for fairly small validation data sizes

### ISSUE THREAD

**Issue by Almad** [Baggz/Amanda#68]
This script

```
amanda = require 'amanda'
assert = require 'assert'

schema =
    type: "array"
    items:
        type: "object"
        properties:
            a: type: "string"
            b: type: "string"
            c: type: "string"
            d: type: "string"
            e: type: "string"
            id: type: "number"
            "bool": type: "boolean"

data   = [
    {"id": 1, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 2, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 3, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 4, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 5, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 6, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 7, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 8, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 9, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 10, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 11, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 12, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 13, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 14, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 15, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 16, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 17, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 18, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 19, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 20, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 21, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 22, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 23, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"},
    {"id": 24, "a": "a", "bool": true, "b": "b", "c": "c", "d": "d", "e": "e"}
]

amanda.validate data, schema, (err) ->
    console.error 'error', err
```

will produce

```
almad@namtar /tmp/xoxo $ coffee amanda-test.coffee 
RangeError: Maximum call stack size exceeded
```

**Comment by lalitkapoor** [comment]
+1

**Comment by exlabs-individual** [comment]
+1

**Comment by OrCharles** [comment]
+1 my payload is nominal

**Comment by passy** [comment]
+1

I don't even reach the point where I see the stack maximum is reached, but it just seems to hang for ever.

**Comment by nenadalm** [comment]
+1 (https://github.com/apiaryio/dredd/issues/65#issuecomment-39737453)

**Comment by nenadalm** [comment]
it seems to be fixed: https://github.com/Baggz/Amanda/pull/32 so why not to merge it?

**Comment by ecordell** [comment]
Added a [pull request in apiaryio/Amanda](https://github.com/apiaryio/Amanda/pull/1) and another in [Baggz/Amanda](https://github.com/Baggz/Amanda/pull/72) to fix this issue

**Comment by Baggz** [comment]
Fixed, #68 has been merged and I’ve published a new version to [NPM](https://www.npmjs.org/package/amanda).
